docs/applications.md

```diff
@@ -175,6 +175,28 @@ An example `data` dictionary to specify on `POST /_matrix/client/r0/pushers/set`
 [APNs documentation]: https://developer.apple.com/library/archive/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/CreatingtheNotificationPayload.html
 
 
+#### VoIP (voice calls and video calls) notifications on iOS
+
+iOS is capable of displaying an on-screen call notification (with answer/deny buttons)
+and playing a ringtone.
+
+However, this requires sending a special kind of push notification.
+Sygnal is not able to send this type of push notification because a VoIP call may
+begin as a result of an encrypted message, so Sygnal has no way to know that a given
+message should initiate a VoIP call.
+
+The current workaround is for your iOS application to register a Notification Service Extension
+that, upon receiving a relevant VoIP notification, makes an HTTP request to Sygnal's
+[`/_matrix/push/v1/notify`](https://spec.matrix.org/latest/push-gateway-api/#post_matrixpushv1notify)
+endpoint in order to trigger the correct type of notification.
+(There is no interface to do this on-device.)
```

Comment:
I'm not 100% convinced of this. See e.g. https://developer.apple.com/documentation/callkit/sending_end-to-end_encrypted_voip_calls

Reply:
That's an excellent find. Wonder if it's new? It doesn't appear on archive.org and the copyright year says 2021.

I will check with the Element iOS team if it would solve their woes in a less hacky way (also probably a more reliable way! :D).

Reply:
>  Wonder if it's new?

https://developer.apple.com/documentation/callkit/cxprovider/3727263-reportnewincomingvoippushpayload says it's available in iOS 14.5+, iPadOS 14.5+, Mac Catalyst 14.5+ . [Apparently](https://en.wikipedia.org/wiki/IOS_version_history#iOS_14_/_iPadOS_14) released end of April this year.

Reply:
I opened an issue for Element iOS to make them aware: https://github.com/vector-im/element-ios/issues/5260

> developer.apple.com/documentation/callkit/cxprovider/3727263-reportnewincomingvoippushpayload says it's available in iOS 14.5+, iPadOS 14.5+, Mac Catalyst 14.5+ . Apparently released end of April this year.

Thanks for looking into it! I'll switch the documentation around to talk about this (maybe mentioning the workaround for historical iOS versions in the meantime for app developers that need to be compatible?).

Reply:
No response thus far so I just documented it as best as I can.